functions/lib/stripe/checkout.py

```diff
@@ -0,0 +1,157 @@
+    logger.error(f"Provided event {event_ref.path} has an organiser {organiser_ref.path} who does not have a stripe account. Returning status=500")
+    return json.dumps({"url": ERROR_URL})
+  
+  # 2b. check if the stripe account is active
+  if (organiser.get("stripeAccountActive") == False):
+    account = stripe.Account.retrieve(organiser.get("stripeAccount"))
+    # if its not active, double check to see if this is the n+1 iteration, hence if they have charges enabled and details submitted for stripe, open their account, else error out
+    if account.charges_enabled and account.details_submitted:
+      transaction.update(organiser_ref, {"stripeAccountActive": True})
+      logger.info(f"Provided organiser {organiser_ref.path} already has all charges enabled and details submitted. Activiating their sportshub stripe account.")
+
+    else:
+      logger.error(f"Provided event {event_ref.path} has an organiser {organiser_ref.path} who does not have a active stripe account. charges_enabled={account.charges_enabled} details_submitted={account.details_submitted} Returning status=500")
+      return json.dumps({"url": ERROR_URL})
+
+  # 3. check again with database in transction in the backend if quantity is still available
+  vacancy = event.get("vacancy")
+
+  if vacancy < quantity:
+    logger.warning(f"Provided event {event_ref.path} does not have enough tickets to fulfill this order. quantity_requested={quantity} vacancy={vacancy}")
+    return json.dumps({"url": cancel_url})
+  
+  # 4. obtain price and organiser stripe account id again from db
+  price = event.get("price")
+  organiser_stripe_account_id = organiser.get("stripeAccount")
+
+  # 4a. check if the price exists for this event
+  if (price == None or not isinstance(price, int) or price <= 1): # we don't want events to be less than stripe fees
+    logger.error(f"Provided event {event_ref.path} does not have a valid price. Returning status=500")
+    return json.dumps({"url": ERROR_URL})
+
+  # 5. set the tickets as sold and reduce vacancy (prevent race condition/ over selling, we will release tickets back after cancelled sale)
+  transaction.update(event_ref, {"vacancy": vacancy - quantity })
+  logger.info(f"Securing {quantity} tickets for event {event_ref.path} at ${price}. There are now {vacancy - quantity} tickets left.")
+
+  # 6. create checkout session with connected account and return link
+  checkout = stripe.checkout.Session.create(
+    mode="payment",
+    line_items=[{
+      "price_data": {
+        "currency": "aud",
+        "product_data": {
+          "name": event.get("name"),
+          "metadata": {
+            "eventId": event_id,
+            "isPrivate": is_private
+          }
+        },
+        "unit_amount": price * 100
+      },
+      "quantity": quantity
+    }],
+    metadata={
+      "eventId": event_id,
+      "isPrivate": is_private
+    },
+    # payment_intent_data={"application_fee_amount": 123},
```

Comment:
remove unnecessary comments

Reply:
this one im gonna keep, as its a good stub code for whenever we want to flick on commission for us. Just to remind us we have the POWERRR MWHAHAHAHAHHAH